cfme/cloud/provider/__init__.py

```diff
@@ -76,7 +76,7 @@ def is_displayed(self):
     including_entities = View.include(VMEntities, use_parent=True)
 
 
-@attr.s(hash=False)
```

Comment:
Can you describe your intent here?

If `cmp=False` and `hash` not set, hash will default to the base class `__hash__` method. Currently we are explicitly defining the providers as _not hashable_, this will change them to use `BaseProvider.__hash__`.

^ Is that the intent? If so, cool.

Reply:
@mshriver, I also had such opinion before looking into this issue. atts is nasty beast and doesn't work as it is expected. 
it doesn't pick up `__hash__`.  I messed around with it for several hours until I managed to make it pick up 
__hash__ and __eq__.